client/client_test.go

```diff
@@ -98,3 +102,316 @@ func TestRedactPasswordArgs(t *testing.T) {
+	}
+}
+
+// TestV2HostnameUnresolvable checks out how the latest client deals with a
+// non-resolvable host name
+func TestV2HostnameUnresolvable(t *testing.T) { // Our test server.
+
+	// a non-existing host name
+	config := &Configuration{
+		Endpoint: "http://non.existing.host.name",
+	}
+
+	gsClient, err := NewV2(config)
+	if err != nil {
+		t.Error(err)
+	}
+
+	responseBody, err := gsClient.CreateAuthToken("email", "password", nil)
+
+	if err == nil {
+		t.Error("Expected error, got nil")
+	}
+	if responseBody != nil {
+		t.Errorf("Expected nil response body, got %#v", responseBody)
+	}
+
+	clientAPIError, ok := err.(*clienterror.APIError)
+	if !ok {
+		t.Error("Type assertion err.(*clienterror.APIError) not successfull")
+	}
+
+	_, ok = clientAPIError.OriginalError.(*net.DNSError)
+	if !ok {
+		t.Error("Type assertion to *net.DNSError not successfull")
+	}
+
+	t.Logf("clientAPIError: %#v", clientAPIError)
+
+	if clientAPIError.ErrorMessage == "" {
+		t.Error("ErrorMessage was empty, expected helpful message.")
+	}
+	if clientAPIError.ErrorDetails == "" {
+		t.Error("ErrorDetails was empty, expected helpful message.")
+	}
+}
+
+// TestV2Timeout tests if the latest client handles timeouts as expected
+func TestV2Timeout(t *testing.T) {
+	// Our test server.
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// enforce a timeout longer than the client's
+		time.Sleep(2 * time.Second)
+		fmt.Fprintln(w, "Hello")
+	}))
+	defer ts.Close()
+
+	clientConfig := &Configuration{
+		Endpoint: ts.URL,
+		Timeout:  500 * time.Millisecond,
+	}
+	gsClient, err := NewV2(clientConfig)
+	if err != nil {
+		t.Error(err)
+	}
+	resp, err := gsClient.CreateAuthToken("email", "password", nil)
+	if err == nil {
+		t.Error("Expected Timeout error, got nil")
+		t.Logf("resp: %#v", resp)
+	} else {
+		clientAPIError, ok := err.(*clienterror.APIError)
+		if !ok {
+			t.Error("Type assertion err.(*clienterror.APIError) not successfull")
+		}
+		if !clientAPIError.IsTimeout {
+			t.Error("Expected clientAPIError.IsTimeout to be true, got false")
+		}
+	}
+}
+
+// TestV2UserAgent tests whether our user-agent header appears in requests
+func TestV2UserAgent(t *testing.T) {
+	clientConfig := &Configuration{
+		UserAgent: "my own user agent/1.0",
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ua := r.Header.Get("user-agent")
+
+		if ua != clientConfig.UserAgent {
+			t.Errorf("Expected '%s', got '%s'", clientConfig.UserAgent, ua)
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{"code": "NONE", "message": "none"}`))
+	}))
+	defer ts.Close()
+
+	clientConfig.Endpoint = ts.URL
+
+	gsClient, err := NewV2(clientConfig)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// just issue a request, don't care about the result
+	_, _ = gsClient.CreateAuthToken("email", "password", nil)
```

Comment:
When result is not to be read from drop the `=`
Just make it `gsClient.CreateAuthToken("email", "password", nil)`

Reply:
changed